mod_ci/controllers.py

```diff
@@ -763,7 +786,57 @@ def progress_reporter(test_id, token):
     return "FAIL"
 
 
+@mod_ci.route('/show_maintenance')
+@login_required
+@check_access_rights([Role.admin])
+@template_renderer('ci/maintenance.html', 404)
+def show_maintenance():
+    return {
+        'modes': MaintenanceMode.query.all()
+    }
+
+
+@mod_ci.route('/toggle_maintenance/<platform>/<status>')
+@login_required
+@check_access_rights([Role.admin])
+def toggle_maintenance(platform, status):
+    db_mode = MaintenanceMode.query.filter(MaintenanceMode.platform ==
+                                           platform).first()
+    if db_mode is None:
+        status = 'failed'
+        message = 'Platform Not found'
+    elif status in ['True', 'False']:
+        db_mode.mode = status
+        g.db.commit()
+        status = 'success'
+        if status == 'True':
+            message = platform + ' platform is in maintenance mode'
+        else:
+            message = platform + ' platform is in active mode'
+    else:
+        status = 'failed'
+        message = 'No Change'
+    return jsonify({
+        'status': status,
+        'message': message
+    })
+
+
 @mod_ci.route('/maintenance-mode/<platform>')
 def in_maintenance_mode(platform):
-    # TODO: fetch status from DB if #85 is implemented
-    return "False"
+    platforms = TestPlatform.values()
+    if platform not in platforms:
+        return 'ERROR'
+    elif platform == 'linux':
```

Comment:
Not future-proof. Each time a platform get's added to TestPlatform, this will need to be updated. There's a method in TestPlatform to go from string to an enum instance.